### PR TITLE
[FIRRTL] Verify OpenAggs, fix agg-of-properties.

### LINKED ISSUE #5494: [FIRRTL] Crash parsing agg-of-properties or non-FieldIDTypeInterface
Sample input:

```firrtl
circuit BundleOfProps:
  module BundleOfProps:
    input x : {a : String}
```

We require all elements of aggregates to also be FieldIDTypeInterface's, should be verifier and/or checked when parsing.

Presently the above crashes.

### PR DESCRIPTION
**Pull request by dtzSiFive**
Add verifiers for both MLIR and FIR + tests.

Fixes #5494 .

### CODE REVIEW

File: lib/Dialect/FIRRTL/FIRRTLTypes.cpp [hunk 1]
```diff
@@ -1741,6 +1744,24 @@ OpenBundleType::getElementTypePreservingConst(size_t index) {
       .Default(type);
 }
 
+auto OpenBundleType::verify(function_ref<InFlightDiagnostic()> emitErrorFn,
+                            ArrayRef<BundleElement> elements, bool isConst)
+    -> LogicalResult {
```
Comment (rwy7) [inline lib/Dialect/FIRRTL/FIRRTLTypes.cpp:1749]: Whats with the trailing return type?

Comment (dtzSiFive) [reply lib/Dialect/FIRRTL/FIRRTLTypes.cpp:1749]: Grabbed it from elsewhere in the file, I can change it :+1:.

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp [hunk 2]
```diff
@@ -923,9 +923,18 @@ ParseResult FIRParser::parseType(FIRRTLType &result, const Twine &message) {
...
+              parseToken(FIRToken::colon, "expected ':' in bundle"))
             return failure();
+          auto loc = getToken().getLoc();
+          if (parseType(type, "expected bundle field type"))
+            return failure();
+
+          // We require that elements of aggregates themselves
+          // support notion of FieldID, reject if the type does not.
+          if (!isa<hw::FieldIDTypeInterface>(type))
+            return emitError(loc, "bundle elements must support fieldID's, "
```
Comment (rwy7) [inline lib/Dialect/FIRRTL/Import/FIRParser.cpp:935]: I don't think the user is going to understand fieldIDs, maybe just leave it at "type 'blah' cannot be used as a field in a bundle"?

Comment (dtzSiFive) [reply lib/Dialect/FIRRTL/Import/FIRParser.cpp:935]: You're totally right, I'll tighten this up, thank you!

(And 100% would be great to be able to render as FIRRTL!

That could be shared with the FIRRTL exporter, maybe, too?)

**Review by rwy7 — APPROVED** [review]
nice

File: lib/Dialect/FIRRTL/FIRRTLTypes.cpp (hunk 1 above)
Comment (rwy7) [reply lib/Dialect/FIRRTL/FIRRTLTypes.cpp]: Gotcha, not a big deal either way :+1:

**Pull request merged**